showcase-docs/dynamic-plugins.md

```diff
@@ -0,0 +1,271 @@
+
+#### About embedding dependencies in the plugin package
+
+The `export-dynamic-plugin` command can also embed some dependencies in the dynamic plugin package, by using the `--embed-package` option. For example:
+
+```bash
+yarn export-dynamic --embed-package @roadiehq/scaffolder-backend-module-http-request
+```
+
+This merges the code of the specified dependencies in the generated dynamic plugin code, and updates the `package.json` file to hoist their dependencies to the top level.
+
+This is useful when wrapping a third-party plugin to make it compatible with the dynamic plugin support, as explained in the next section.
+
+#### Wrapping a third-party plugin to add dynamic plugin support
+
+In order to add dynamic plugin support to a third-party plugin, without touching the third-party plugin source code, a wrapper plugin can be created that will:
+
+- import the third-party plugin,
+- include the additions described above,
+- export it as a dynamic plugin.
+
+An example of such a wrapper plugin can be found in the [dynamic backend plugin showcase repository](https://github.com/janus-idp/dynamic-backend-plugins-showcase/tree/main/plugins/scaffolder-backend-module-http-request-wrapped): it wraps the `@roadiehq/scaffolder-backend-module-http-request` package to make it compatible with the dynamic plugin support and, through the use of the `--embed-package` option in the [`export-dynamic` script](https://github.com/janus-idp/dynamic-backend-plugins-showcase/blob/d254c065764ab49289c2bcaad9fd996d49003f9d/plugins/scaffolder-backend-module-http-request-wrapped/package.json#L21), embeds the wrapped plugin code in the generated code and hoist its `@backstage` dependencies as peer dependencies in the resulting dynamic plugin.
+
+## Installing a dynamic plugin package in the showcase
+
+### Local configuration
+
+- Create a `dynamic-plugins-root` folder at the root of the showcase application repository.
+
+- In the `app-config.yaml`` file, add the following entry:
+
+  ```yaml
+  dynamicPlugins:
+    rootDirectory: dynamic-plugins-root
+  ```
+
+- Copy the dynamic plugin package to the `dynamic-plugins-root` folder with the following commands:
+
+  ```bash
+  pkg=<local dist-dynamic sub-folder or remote package name of the dynamic plugin package>
+  archive=$(npm pack $pkg)
+  tar -xzf "$archive" && rm "$archive"
+  mv package $(echo $archive | sed -e 's:\.tgz$::')
+  ```
+
+  It will create a sub-folder named after the package name, and containing the dynamic plugin package.
+
+- Start the showcase application. During the initialization step it should have a log entry similar to the following:
+
+  ```
+  backstage info loaded dynamic backend plugin '@scope/some-plugin-dynamic' from 'file:///showacase-root/dynamic-plugins-root/scope-some-plugin-dynamic-0.0.1'
```

Comment:
```suggestion
  backstage info loaded dynamic backend plugin '@scope/some-plugin-dynamic' from 'file:///showacase-root/dynamic-plugins/scope-some-plugin-dynamic-0.0.1'
```

The dockerfile uses dynamic-plugins

Reply:
see answer above: https://github.com/janus-idp/backstage-showcase/pull/522#discussion_r1344704182